Configure native dependencies from the makefile.

pchote opened the pull request:
Followup fixes to #7953.

pchote committed:
Configure native dependencies from the makefile.

Phrohdoh commented on configure:
```diff
@@ -5,6 +5,11 @@ os=`uname`
 	sonames=(liblua.so.5.1.5 liblua5.1.so.5.1 liblua5.1.so.0 liblua.so.5.1 liblua-5.1.so liblua5.1.so)
+
+	if [ -f Eluant.dll.config ]; then
+		exit 0
```
Success if no config?

pchote replied:
Success if config.

Phrohdoh replied:
Oh, of course. That's an embarrassing mistake.

Phrohdoh commented on configure:
```diff
@@ -22,9 +28,18 @@ if [ "$os" == 'Linux' ]; then
+	fi
+
+	if [ ! -f liblua.5.1.dylib ]; then
+		echo "Fetching OSX lua library from GitHub."
```
`OS X` is preferred.

pchote replied:
Good catch.

Phrohdoh commented:
:+1:

pchote commented:
Fixed, and added a commit that moves `configure` &rarr; `thirdparty/configure-native-deps.sh` to address confusion/complaints on IRC about our nonstandard `configure` usage.  This command no longer needs to be run manually.

Mailaender commented:
This is a very good idea. I saw distribution packagers getting confused and running our script with their usual %configure macros. Also it was already redundant to run, because the Makefile executed it, too for convenience.

There is just a problem that you made the script non-executable (chmod 0644 from 0755) which you will need to revert to not make this fail:

```
make native-dependencies
make: execvp: ./thirdparty/configure-native-deps.sh: Insufficient rights
Makefile:305: recipe for target 'native-dependencies' failed
make: *** [native-dependencies] Error 127
```

pchote committed:
Move native deps script into thirdparty to avoid confusion.

pchote commented:
This appears to be a git bug.  The permissions are correct in my working copy, and discarding and recommiting the files (with verified good permissions) still shows 755 &rarr; 644 when pushed.

```
paul@localhost:thirdparty+more-dependency-fixes$ ls -lah *.sh
-rwxr-xr-x  1 paul  staff   1.5K  3 May 14:48 configure-native-deps.sh
-rwxr-xr-x  1 paul  staff   1.0K  3 May 14:47 fetch-thirdparty-deps-windows.sh
-rwxr-xr-x  1 paul  staff   2.5K  3 May 14:48 fetch-thirdparty-deps.sh
```

I'm not sure how to proceed from here.

Mailaender commented:
I seem to be unaffected by that bug: see https://github.com/OpenRA/OpenRA/pull/8082.
